### PR TITLE
Move track to its own package

### PR DESCRIPTION
**Pull request by edwardrf**

### CODE REVIEW

File: src/cmd/cli/command/commands.go [hunk 1]
```diff
@@ -32,6 +33,10 @@ import (
 const authNeeded = "auth-needed" // annotation to indicate that a command needs authorization
 var authNeededAnnotation = map[string]string{authNeeded: ""}
 
+func P(name string, value interface{}) cliClient.Property {
+	return cliClient.Property{Name: name, Value: value}
+}
+
```
Comment (edwardrf) [inline src/cmd/cli/command/commands.go:39]: A little copying to avoid circular dependency

File: src/cmd/cli/command/commands.go [hunk 2]
```diff
@@ -1021,3 +1029,7 @@ func awsInEnv() bool {
 func doInEnv() bool {
 	return os.Getenv("DIGITALOCEAN_ACCESS_TOKEN") != "" || os.Getenv("DIGITALOCEAN_TOKEN") != ""
 }
+
+func IsCompletionCommand(cmd *cobra.Command) bool {
+	return cmd.Name() == cobra.ShellCompRequestCmd || (cmd.Parent() != nil && cmd.Parent().Name() == "completion")
+}
```
Comment (edwardrf) [inline src/cmd/cli/command/commands.go:1035]: Intentional copying to avoid dependency (circle)

**Review by lionello — APPROVED** [review]

File: src/pkg/track/track.go [hunk 3]
```diff
@@ -1,53 +1,58 @@
...
+	go func() {
 		defer trackWG.Done()
-		_ = client.Track(name, props...)
-	}(client)
+		if Fabric == nil {
+			term.Debugf("No FabricClient to track event: %v", name)
+			return
+		}
+		Fabric.Track(name, props...)
+	}()
```
Comment (lionello) [inline src/pkg/track/track.go:38]: pass `Fabric` as arg

Comment (edwardrf) [reply src/pkg/track/track.go]: Great call!

File: src/pkg/cli/connect.go [hunk 4]
```diff
@@ -69,6 +70,7 @@ func Connect(cluster string, loader client.ProjectLoader) client.GrpcClient {
 
 func NewClient(ctx context.Context, cluster string, providerID client.ProviderID, loader client.ProjectLoader) (client.FabricClient, client.Provider) {
 	grpcClient := Connect(cluster, loader)
+	track.Fabric = grpcClient // Update track client
```
[the remark this answers is not in the document]
Comment (edwardrf) [reply src/pkg/cli/connect.go:73]: No, this is inline with the current behavior, where the global `client` variable gets updated by the `NewClient` call returned value, and if the track happens before it, an lazily initialized client with `cluster` is created.
Update the client here to make sure we get an updated access token after login.

**Pull request merged**